packages/core/src/App/Containers/Layout/header/default-header.tsx

```diff
@@ -0,0 +1,132 @@
+    const {
+        currency,
+        is_bot_allowed,
+        is_dxtrade_allowed,
+        is_logged_in,
+        is_logging_in,
+        is_mt5_allowed,
+        is_switching,
+        is_landing_company_loaded,
+    } = client;
+    const { app_routing_history, current_language, platform } = common;
+    const { addNotificationMessage, client_notifications, removeNotificationMessage } = notifications;
+    const { setTogglePlatformType } = traders_hub;
+    const {
+        header_extension,
+        is_desktop,
+        is_mobile,
+        is_app_disabled,
+        is_route_modal_on,
+        is_trading_assessment_for_existing_user_enabled,
+    } = ui;
+
+    const history = useHistory();
+
+    const addUpdateNotification = () => addNotificationMessage(client_notifications?.new_version_available);
+    const removeUpdateNotification = React.useCallback(
+        () => removeNotificationMessage({ key: 'new_version_available' }),
+        [removeNotificationMessage]
+    );
+
+    React.useEffect(() => {
+        document.addEventListener('IgnorePWAUpdate', removeUpdateNotification);
+        return () => document.removeEventListener('IgnorePWAUpdate', removeUpdateNotification);
+    }, [removeUpdateNotification]);
+
+    const onClickDeposit = () => history.push(routes.cashier_deposit);
+
+    const filterPlatformsForClients = payload =>
+        payload.filter(config => {
+            if (config.link_to === routes.mt5) {
+                return !is_logged_in || is_mt5_allowed;
+            }
+            if (config.link_to === routes.dxtrade) {
+                return is_dxtrade_allowed;
+            }
+            if (
+                config.link_to === routes.bot ||
+                config.href === routes.binarybot ||
+                config.href === routes.smarttrader
+            ) {
+                return is_bot_allowed;
+            }
+            return true;
+        });
+
+    return (
+        <header
+            className={classNames('header', {
+                'header--is-disabled': is_app_disabled || is_route_modal_on,
+                'header--is-hidden': platforms[platform],
+            })}
+        >
+            <div className='header__menu-items'>
+                <div className='header__menu-left'>
+                    {is_desktop && (
+                        <PlatformSwitcher
+                            app_routing_history={app_routing_history}
+                            is_landing_company_loaded={is_landing_company_loaded}
+                            is_logged_in={is_logged_in}
+                            is_logging_in={is_logging_in}
+                            platform_config={filterPlatformsForClients(platform_config)}
+                            setTogglePlatformType={setTogglePlatformType}
+                            current_language={current_language}
+                        />
+                    )}
+                    {is_mobile && (
```

Comment:
*NIT*: Can we use MobileWrapper here?. This eliminated `is_mobile` condition and `React.Fragment`

Reply:
i thought, while using store, we should be using is_mobile and is_desktop too from store throughout the file. wouldn't mixing is_mobile and MobileWrapper make code ugly?